src/components/editor-page/renderer-pane/render-iframe.tsx

```diff
@@ -105,15 +107,15 @@ export const RenderIframe: React.FC<RenderIframeProps> = ({
     useCallback(() => onMakeScrollSource?.(), [onMakeScrollSource])
   )
 
+  const eventEmitter = useExtensionEventEmitter()
+
   useEditorReceiveHandler(
-    CommunicationMessageType.ON_TASK_CHECKBOX_CHANGE,
-    useCallback(
-      (values: OnTaskCheckboxChangeMessage) => {
-        const lineOffset = getGlobalState().noteDetails.frontmatterRendererInfo.lineOffset
-        onTaskCheckedChange?.(values.lineInMarkdown + lineOffset, values.checked)
-      },
-      [onTaskCheckedChange]
-    )
+    CommunicationMessageType.EXTENSION_EVENT,
+    useMemo(() => {
+      return eventEmitter === undefined
+        ? undefined
+        : (values: ExtensionEvent) => eventEmitter?.emit(values.eventName, values.payload)
```

Comment:
```suggestion
        : (values: ExtensionEvent) => eventEmitter!.emit(values.eventName, values.payload)
```
At this point we know eventEmitter is not undefined